src/media/MediaAnnotator.ts

```diff
@@ -0,0 +1,132 @@
+import { Unsubscribe } from 'redux';
+import BaseAnnotator, { Options } from '../common/BaseAnnotator';
+import PopupManager from '../popup/PopupManager';
+import { DrawingManager } from '../drawing';
+import { RegionCreationManager, RegionManager } from '../region';
+import { CreatorStatus, getCreatorStatus } from '../store/creator';
+import { getAnnotation, getFileId, getIsCurrentFileVersion, getRotation } from '../store';
+import { Manager } from '../common/BaseManager';
+import './MediaAnnotator.scss';
+import { MEDIA_LOCATION_INDEX, FRAME } from '../constants';
+
+export const CSS_IS_DRAWING_CLASS = 'ba-is-drawing';
+
+export default class MediaAnnotator extends BaseAnnotator {
+    annotatedEl?: HTMLElement;
+
+    managers: Set<Manager> = new Set();
+
+    storeHandler?: Unsubscribe;
+
+    constructor(options: Options) {
+        super(options);
+
+        this.storeHandler = this.store.subscribe(this.handleStore);
+    }
+
+    destroy(): void {
+        if (this.storeHandler) {
+            this.storeHandler();
+        }
+
+        if (this.managers) {
+            this.managers.forEach(manager => manager.destroy());
+        }
+
+        super.destroy();
+    }
+
+    getAnnotatedElement(): HTMLElement | null | undefined {
+        return this.containerEl?.querySelector('.bp-media');
+    }
+
+
+    getManagers(parentEl: HTMLElement, referenceEl: HTMLVideoElement): Set<Manager> {
+        const fileId = getFileId(this.store.getState());
+        const isCurrentFileVersion = getIsCurrentFileVersion(this.store.getState());
+        const resinTags = { fileid: fileId, iscurrent: isCurrentFileVersion };
+        this.managers.forEach(manager => {
+            if (!manager.exists(parentEl)) {
+                manager.destroy();
+                this.managers.delete(manager);
+            }
+        });
+
+        if (this.managers.size === 0) {
+            this.managers.add(new PopupManager({ location: MEDIA_LOCATION_INDEX,   referenceEl, resinTags, targetType : FRAME}));
+            this.managers.add(new DrawingManager({ location: MEDIA_LOCATION_INDEX, referenceEl, resinTags, targetType : FRAME}));
+            this.managers.add(new RegionManager({ location: MEDIA_LOCATION_INDEX, referenceEl, resinTags, targetType : FRAME}));
+            this.managers.add(new RegionCreationManager({ location: MEDIA_LOCATION_INDEX, referenceEl, resinTags, targetType : FRAME}));
+        }
+
+        return this.managers;
+    }
+
+  
+
+    // returns the actual video element
+    getReference(): HTMLVideoElement | null | undefined {
```

Comment:
Can the return type be inferred from what is being returned?

Reply:
No not in this case. Removing that causes a TS error